### PR TITLE
feat(charts): added PF dark theme support

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly-react/issues/7188

Preview link - https://patternfly-react-pr-7310.surge.sh/charts/area-chart/

To enable dark theme, you need to add `class="pf-theme-dark"` to the `<html>` element. You can do this manually in dev tools, or paste this into the dev tools console to toggle the class on/off:

```
document.documentElement.classList.toggle('pf-theme-dark');
```

The chart (and component, layout, etc) examples have a default background set to PF's primary background color. To disable that and see the charts against the main area background color (PF's secondary background color), either manually update the `.ws-preview` class in dev tools to have no background or one set to `--pf-global--BackgroundColor--200`, or you can paste this in the dev tools console:

```
const wsp = document.querySelectorAll('.ws-preview');
for (let i = 0; i < wsp.length; i++) {
  wsp[i].style.background = 'none';
}
```

### CODE REVIEW

**Comment by mcoker** [comment]
Current screenshots:
Screen Shot 2022-05-02 at 10 29 03 AM
Screen Shot 2022-05-02 at 10 29 05 AM
Screen Shot 2022-05-02 at 10 29 11 AM
Screen Shot 2022-05-02 at 10 29 14 AM
Screen Shot 2022-05-02 at 10 29 22 AM
Screen Shot 2022-05-02 at 10 29 25 AM
Screen Shot 2022-05-02 at 10 29 32 AM
Screen Shot 2022-05-02 at 10 29 36 AM
Screen Shot 2022-05-02 at 10 29 39 AM
Screen Shot 2022-05-02 at 10 29 43 AM
Screen Shot 2022-05-02 at 10 29 48 AM
Screen Shot 2022-05-02 at 10 29 50 AM
Screen Shot 2022-05-02 at 10 29 57 AM
Screen Shot 2022-05-02 at 10 30 00 AM
Screen Shot 2022-05-02 at 10 30 03 AM
Screen Shot 2022-05-02 at 10 30 09 AM
Screen Shot 2022-05-02 at 10 30 11 AM
Screen Shot 2022-05-02 at 10 30 13 AM
Screen Shot 2022-05-02 at 10 30 17 AM
Screen Shot 2022-05-02 at 10 30 19 AM
Screen Shot 2022-05-02 at 10 30 22 AM
Screen Shot 2022-05-02 at 10 30 24 AM
Screen Shot 2022-05-02 at 10 30 29 AM
Screen Shot 2022-05-02 at 10 30 31 AM
Screen Shot 2022-05-02 at 10 30 34 AM
Screen Shot 2022-05-02 at 10 30 36 AM
Screen Shot 2022-05-02 at 10 30 44 AM
Screen Shot 2022-05-02 at 10 30 46 AM
Screen Shot 2022-05-02 at 10 30 50 AM
Screen Shot 2022-05-02 at 10 31 20 AM
Screen Shot 2022-05-02 at 10 31 24 AM
Screen Shot 2022-05-02 at 10 31 29 AM
Screen Shot 2022-05-02 at 10 31 35 AM
Screen Shot 2022-05-02 at 10 31 37 AM
Screen Shot 2022-05-02 at 10 31 40 AM
Screen Shot 2022-05-02 at 10 31 44 AM
Screen Shot 2022-05-02 at 10 31 46 AM
Screen Shot 2022-05-02 at 10 31 48 AM
Screen Shot 2022-05-02 at 10 31 54 AM
Screen Shot 2022-05-02 at 10 31 59 AM
Screen Shot 2022-05-02 at 10 32 01 AM
Screen Shot 2022-05-02 at 10 32 03 AM
Screen Shot 2022-05-02 at 10 32 05 AM

This is a custom chart with multiple positive/negative measures to show the palette contrast.

Screen Shot 2022-05-02 at 10 34 01 AM

File: packages/react-charts/src/components/ChartLabel/ChartLabel.tsx [hunk 1]
```diff
@@ -224,6 +224,7 @@ export const ChartLabel: React.FunctionComponent<ChartLabelProps> = ({
         textAnchor // textAnchor prop must override given theme styles
       },
       {
+        // fill: ChartCommonStyles.label.fill,
```
Comment (dlabrecq) [inline packages/react-charts/src/components/ChartLabel/ChartLabel.tsx:227]: This comment can be removed

File: packages/react-charts/src/components/ChartTheme/themes/base-theme.ts [hunk 2]
```diff
@@ -149,46 +150,43 @@ export const BaseTheme = {
...
-        stroke: chart_axis_tick_stroke_Color.value,
+        stroke: chart_axis_tick_stroke_Color.var,
         strokeLinecap: STROKE_LINE_CAP,
         strokeLinejoin: STROKE_LINE_JOIN,
         strokeWidth: chart_axis_tick_Width.value
       },
-      tickLabels: {
-        ...LABEL_PROPS,
-        fill: chart_axis_tick_label_Fill.value
-      }
```
Comment (dlabrecq) [inline packages/react-charts/src/components/ChartTheme/themes/base-theme.ts:181]: Not sure this `fill` should be removed? 

The `fill` had a value of #4f5255. Not sure what `chart_global_label_Fill.va`r is?

File: packages/react-charts/src/components/ChartTheme/themes/dark/multi-color-ordered-theme.ts [hunk 3]
```diff
@@ -29,31 +29,31 @@ import { ColorTheme } from '../color-theme';
...
+  chart_color_blue_200.var,
+  chart_color_green_400.var,
+  chart_color_cyan_200.var,
+  chart_color_gold_200.var,
+  chart_color_orange_400.var,
+  chart_color_blue_400.var,
+  chart_color_green_200.var,
+  chart_color_cyan_400.var,
+  chart_color_gold_400.var,
+  chart_color_orange_200.var
```
Comment (dlabrecq) [inline packages/react-charts/src/components/ChartTheme/themes/dark/multi-color-ordered-theme.ts:56]: Michael Celedonia suggested he was having trouble with color pallets and the dark them. I understood he wanted to shift the multi-color pallet by one color.

Are we not going to create vars for all color pallets? 

I see that `chart_bullet_negative_measure_ColorScale_100.var` was used for the bullet chart

File: packages/react-charts/src/components/ChartTheme/themes/bullet-theme.ts [hunk 4]
```diff
@@ -108,11 +108,11 @@ export const BulletPrimaryDotMeasureTheme = {
...
-      chart_color_red_100.value,
-      chart_color_red_200.value,
-      chart_color_red_300.value,
-      chart_color_red_400.value,
-      chart_color_red_500.value
+      chart_bullet_negative_measure_ColorScale_100.var,
+      chart_bullet_negative_measure_ColorScale_200.var,
+      chart_bullet_negative_measure_ColorScale_300.var,
+      chart_bullet_negative_measure_ColorScale_400.var,
+      chart_bullet_negative_measure_ColorScale_500.var
```
Comment (dlabrecq) [inline packages/react-charts/src/components/ChartTheme/themes/bullet-theme.ts:115]: Are we going to use vars for all color pallets? 

Ideally, all color pallets would be implemented similarly. If not, then perhaps this should be reverted?

**Review by dlabrecq — CHANGES_REQUESTED** [review]
The "Cannot find module" won't succeed until the PatternFly core package is updated.

Alternatively, the new vars for the bullet chart color pallet could be reverted? Doesn't seem like we have vars for all color pallets.

**Review by mceledonia — APPROVED** [review]

**Review by dlabrecq — COMMENTED** [review]
Tested locally and within Cost Management and the UIs look good. Not set up to test the dark theme in Cost, tho.

I noticed a few more places where we still reference value for colors

Screen Shot 2022-05-07 at 1 54 28 PM

**Comment by mcoker** [comment]
@dlabrecq updated those spots to use `var`

**Review by dlabrecq — APPROVED** [review]
LGTM

**Review by tlabaj — APPROVED** [review]
LGTM

**Pull request merged**